pwiz_tools/Skyline/Test/SrmSettingsTest.cs

```diff
@@ -1140,6 +1141,24 @@ TransitionIonMobilityFiltering CheckIonMobilitySettingsBackwardCompatibility(str
         }
 
 
+        [TestMethod]
+        public void SerializeProteinAssociationSettingsTest()
+        {
+            const string proteinAssociationSerialized = "<protein_association min_peptides_per_protein=\"3\" " +
+                                                        "group_proteins=\"true\" " +
+                                                        "find_minimal_protein_list=\"true\" " +
+                                                        "remove_subset_proteins=\"true\" " +
+                                                        "shared_peptides=\"AssignedToBestProtein\" />";
+            AssertEx.DeserializeNoError<ProteinAssociation.ParsimonySettings>(proteinAssociationSerialized, DocumentFormat.VERSION_22_13);
+
+            var parsimonySettings = AssertEx.Deserialize<ProteinAssociation.ParsimonySettings>(proteinAssociationSerialized);
+            Assert.AreEqual(3, parsimonySettings.MinPeptidesPerProtein);
+            Assert.AreEqual(true, parsimonySettings.GroupProteins);
+            Assert.AreEqual(true, parsimonySettings.FindMinimalProteinList);
+            Assert.AreEqual(true, parsimonySettings.RemoveSubsetProteins);
+            Assert.AreEqual(ProteinAssociation.SharedPeptides.AssignedToBestProtein, parsimonySettings.SharedPeptides);

```

Comment:
I would add a bit more here. Does the class do any validation at all? If so, you should check DeserializeError() for out of bounds value, and maybe at least test min_peptides_per_protein="0".

Reply:
No validation yet but I'll add it. Why 0 specifically?

Reply:
Often testers will test the acceptable edges of allowed values. Zero is the lowest this should be allowed to go, but off-by-ones are common and if zero became unallowed that would be an important bug. A professional tester should be able to tick off: 1) allowed values, 2) edge case allowed values, 3) error cases.